diff --git a/debian/control b/debian/control
index 1eee9046..37005804 100644
--- a/debian/control
+++ b/debian/control
@@ -1,4 +1,4 @@
-Source: ignition
+Source: ignition-legacy
 Maintainer: Debian Cloud Team <debian-cloud@lists.debian.org>
 Uploaders: Michael Banck <mbanck@debian.org>
 Section: golang
@@ -8,37 +8,19 @@ Build-Depends: debhelper-compat (= 13),
                dh-exec,
                dh-golang,
                golang-any,
-               golang-github-aws-aws-sdk-go-dev,
-               golang-github-beevik-etree-dev,
-               golang-github-coreos-go-json-dev,
-               golang-github-coreos-go-systemd-dev,
-               golang-github-coreos-vcontext-dev,
-               golang-github-google-renameio-dev,
-               golang-github-google-uuid-dev,
-               golang-github-pin-tftp-dev,
-               golang-github-spf13-pflag-dev (>= 1.0.6~git20210604),
-               golang-github-stretchr-testify-dev,
-               golang-github-vincent-petithory-dataurl-dev,
-               golang-github-vmware-vmw-guestinfo-dev (>= 0.0~git20220317) [amd64 i386],
-               golang-github-vmware-vmw-ovflib-dev,
-               golang-go-semver-dev,
-               golang-golang-x-net-dev,
-               golang-golang-x-oauth2-google-dev,
-               golang-golang-x-sys-dev,
-               golang-google-api-dev,
-               golang-google-cloud-dev,
                libblkid-dev
 Standards-Version: 4.6.0
 Vcs-Browser: https://salsa.debian.org/cloud-team/ignition
 Vcs-Git: https://salsa.debian.org/cloud-team/ignition.git
 Homepage: https://github.com/coreos/ignition
 Rules-Requires-Root: no
-XS-Go-Import-Path: github.com/coreos/ignition
+XS-Go-Import-Path: github.com/coreos/ignition/v2

-Package: ignition
+Package: ignition-legacy
 Architecture: any
 Depends: ${misc:Depends},
          ${shlibs:Depends}
+Conflicts: ignition
 Built-Using: ${misc:Built-Using}
 Description: First boot installer and configuration tool
  Ignition is the utility invented for Fedora CoreOS and RHEL CoreOS to
